File: client/src/services/dtos/chat-conversation-dto.ts
import { UserDto } from './user-dto';

interface ConversationReadStatusDto {
  readonly readBy: UserDto;
  readonly readAt: string;
  readonly latestMessagesRead: boolean;
}

export interface ChatConversationDto {
  readonly muted: boolean;
  readonly id: number;
  readonly participants: readonly UserDto[];
  readonly read: boolean;
  readonly created: Date;
  readonly latestChatMessageId: number;
  readonly readStatuses: readonly ConversationReadStatusDto[];
  readonly restricted: boolean;
  readonly unavailable: boolean;
  readonly blocked: boolean;
  readonly mustFollowToChat: boolean;
  readonly needConfirmationBeforeSendingMessage: boolean;
}

File: back-end/src/main/java/com/huyphan/dtos/ChatConversationDto.java
package com.huyphan.dtos;

import java.util.List;
import javax.persistence.Transient;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ChatConversationDto {

    private long id;

    private List<UserDto> participants;

    private List<ConversationReadStatusDto> readStatuses;

    private Long latestChatMessageId;

    private String created;

    private boolean muted;

    private boolean restricted;

    private boolean unavailable;

    private boolean blocked;

    private boolean mustFollowToChat;
}
